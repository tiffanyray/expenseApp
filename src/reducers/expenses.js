

// Expenses Reducer 
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [
        ...state,
        action.expense
      ]
    case 'REMOVE_EXPENSE' :
      return state.filter( ({ id }) => id !== action.id);
    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        if(expense.id === action.id) {
          return { 
            ...expense, //spread operator that makes it easy to 
            ...action.updates //update any of the existing infomation provided
          }
        } else {
          return expense;
        }
      })
    default: 
      return state;
  }
};

export default expensesReducer;