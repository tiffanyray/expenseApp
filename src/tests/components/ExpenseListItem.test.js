import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';
//setting up screenshot for individual expense list item
test('should render expense list item component with fixture data', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});