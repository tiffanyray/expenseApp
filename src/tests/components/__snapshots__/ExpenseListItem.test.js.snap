// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item component with fixture data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      gum
    </h3>
  </Link>
  <p>
    199
     - 
    0
  </p>
</div>
`;
